android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsCommon.java

```diff
@@ -178,7 +178,7 @@ public static AdManagerAdRequest buildAdRequest(ReadableMap adRequestOptions) {
 
       for (Map.Entry<String, Object> entry : customTargeting.entrySet()) {
         String key = entry.getKey();
-        String value = (String) entry.getValue();
+        ArrayList<String> value = (ArrayList<String>) entry.getValue();
```

Comment:
I'm having a difficult time understanding how this does not just convert the crash from being:

"when passing array of strings it crashes" to

"when passing single string it crashes"

That is, this just makes it crash for the other case now. It appears both types of input should be accepted (`String` and `ArrayList<String>`) and you need to do a runtime check like `if (value instanceof String) ...` with conditional cast

Reply:
Yes, I think so, too.

Reply:
Just a note @AbdelhalimAhmed that this looks close to going in, but this comment must be addressed as it appears it will cause a crash in the opposite direction of your fix, and both cases need to be handled...